src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_FindReferences_Current.cs

```diff
@@ -27,63 +27,62 @@ public static partial class SymbolFinder
         {
             using (Logger.LogBlock(FunctionId.FindReference, cancellationToken))
             {
-                var handled = await TryFindReferencesInServiceProcessAsync(
-                    symbolAndProjectId, solution, progress,
-                    documents, options, cancellationToken).ConfigureAwait(false);
-                if (handled)
+                // If ProjectId is null then this is a call through our old public API.  We don't have
+                // the necessary data to effectively run the call out of proc.
+                if (symbolAndProjectId.ProjectId != null)
                 {
-                    return;
+                    var client = await RemoteHostClient.TryGetClientAsync(solution.Workspace, cancellationToken).ConfigureAwait(false);
+                    if (client != null)
+                    {
+                        // Create a callback that we can pass to the server process to hear about the 
+                        // results as it finds them.  When we hear about results we'll forward them to
+                        // the 'progress' parameter which will then update the UI.
+                        var serverCallback = new FindReferencesServerCallback(solution, progress, cancellationToken);
+
+                        var success = await client.TryRunRemoteAsync(
+                            WellKnownServiceHubServices.CodeAnalysisService,
+                            nameof(IRemoteSymbolFinder.FindReferencesAsync),
+                            solution,
+                            new object[]
+                            {
+                                SerializableSymbolAndProjectId.Dehydrate(symbolAndProjectId),
+                                documents?.Select(d => d.Id).ToArray(),
+                                SerializableFindReferencesSearchOptions.Dehydrate(options),
+                            },
+                            cancellationToken,
+                            serverCallback).ConfigureAwait(false);
+
+                        if (success)
+                        {
+                            return;
+                        }
+                    }
                 }
 
-                // Couldn't effectively search using the OOP process.  Just perform the search in-proc.
+                // Couldn't effectively search in OOP. Perform the search in-proc.
                 await FindReferencesInCurrentProcessAsync(
-                    symbolAndProjectId, solution, progress,
-                    documents, options, cancellationToken).ConfigureAwait(false);
+                    symbolAndProjectId,
+                    solution,
+                    progress,
+                    documents,
+                    options,
+                    cancellationToken).ConfigureAwait(false);
```

Comment:
can this be undone?

Reply:
What do you want to be undone? The code now follows the same pattern as other remote invocations described above.

Reply:
wrapping every arg to a new line.  It doesn't seem like anything else changed there.  but it's up to you here.

Reply:
Sure, I'll do that.